modules/fuse-patch/patch-core/src/main/java/org/jboss/fuse/patch/impl/PatchServiceImpl.java

```diff
@@ -1619,14 +1613,10 @@ private boolean refreshAutoDeployCache(File[] zips) {
 
     @Override
     public void undeploy(Patch patch) {
-        for (Iterator<DeployedPatch> iterator = deployedPatches.values().iterator(); iterator.hasNext(); ) {
-            DeployedPatch dp = iterator.next();
-            if (dp.getPatchData().getId().equals(patch.getPatchData().getId())) {
-                iterator.remove();
```

Comment:
@VratislavHais I had to revert this particular change - I needed this `iterator.remove()`.